src/ray/raylet/scheduling/scheduling_policy.cc

```diff
@@ -204,5 +204,37 @@ int64_t SchedulingPolicy::HybridPolicy(const ResourceRequest &resource_request,
                                 require_available, is_node_available);
 }
 
+int64_t SchedulingPolicy::RandomPolicy(const ResourceRequest &resource_request,
+                                       std::function<bool(int64_t)> is_node_available) {
+  int64_t best_node = -1;
+  if (nodes_.empty()) {
+    return best_node;
+  }
+
+  std::uniform_int_distribution<int> distribution(0, nodes_.size() - 1);
+  int idx = distribution(gen_);
+  auto iter = std::next(nodes_.begin(), idx);
+  for (size_t i = 0; i < nodes_.size(); ++i) {
+    // TODO(scv119): if there are a lot of nodes died or can't fulfill the resource
```

Comment:
Isn't this expected? (is it possible to fix this? )

Reply:
yeah I think it's fixable; a dumb solution would first filter out nodes that doesn't meet the requirement and then pick a node from them randomly.

Reply:
Ah, I see. It makes sense.

Reply:
Fun single-pass leetcode style algorithm to solve this: 

pick the nth alive node with probability 1/n. After you've finished iterating over the nodes, you will have picked one with uniform random probability